### PR TITLE
Adds Outcome documentation

### PR DESCRIPTION
**Pull request by cwmyers**

### CODE REVIEW

File: lib/src/main/kotlin/app/cash/quiver/Outcome.kt [hunk 1]
```diff
@@ -24,11 +24,49 @@ import kotlin.experimental.ExperimentalTypeInference
...
+ *
+ * ```kotlin
+ * data class Present<A>(val value: A) : Outcome<Nothing, A>
+ * data class Failure<E>(val error: E) : Outcome<E, Nothing>
+ * object Absent : Outcome<Nothing, Nothing>
+ * ```
+ *
+ * or you can use the extension methods thusly:
+ *
+ * ```kotlin
```
Comment (cwmyers) [inline lib/src/main/kotlin/app/cash/quiver/Outcome.kt:37]: Do I need the kotlin annotation here?  I'm not sure how dokka works.

File: lib/src/main/kotlin/app/cash/quiver/Outcome.kt [hunk 2]
```diff
@@ -115,6 +201,13 @@ inline fun <E, A, B, C, D, EE, F> Outcome<E, A>.zip(
     o4.map { e -> f(a, b, c, d, e) }
   }.flatten()
 
+/**
+ * An extension method on Either<E, Option<A>> that converts it to an Outcome<E, A>.
+ *
+ * ```
```
Comment (cwmyers) [inline lib/src/main/kotlin/app/cash/quiver/Outcome.kt:207]: oops

File: lib/src/main/kotlin/app/cash/quiver/Outcome.kt (hunk 1 above)
Comment (hugomd) [reply lib/src/main/kotlin/app/cash/quiver/Outcome.kt:37]: image

I _think_ it defaults to kotlin. No harm in including it here to be explicit 👍🏻

File: lib/src/main/kotlin/app/cash/quiver/Outcome.kt [hunk 3]
```diff
@@ -24,11 +24,49 @@ import kotlin.experimental.ExperimentalTypeInference
...
+ *
+ * or you can use the extension methods thusly:
+ *
+ * ```kotlin
+ * A.present()
+ * E.failure()
+ * ```
+ *
+ * You can also easily convert an `Either<Option<A>>` to an Outcome using `toOutcome()`
```
Comment (hugomd) [inline lib/src/main/kotlin/app/cash/quiver/Outcome.kt:42]: nit: `toOutcome()` can link directly to the documentation page for `toOutcome()`: 
```suggestion
 * You can also easily convert an `Either<Option<A>>` to an Outcome using  [`toOutcome()`](app.cash.quiver.toOutcome)
```

Comment (hugomd) [reply lib/src/main/kotlin/app/cash/quiver/Outcome.kt]: This can probably be a separate pull request to add links across all of our documentation

**Review by hugomd — APPROVED** [review]
Excellent ! 📝

**Pull request merged**